docs/data/sql_functions.yml

```diff
@@ -1219,6 +1219,35 @@ valueconstruction:
   - table: NUMERIC.rows
     description: Creates a NUMERIC interval of rows (commonly used in window creation).
 
+valuemodification:
+  - sql: OBJECT_UPDATE(object, key, value [, key, value , ...])
+    table: OBJECT.objectUpdate(key, value [, key, value , ...])
+    description: |
+      Updates existing fields in a structured object by providing key-value pairs.
```

Comment:
I am curious:
- I assume that arrays are not included in this 
- are there restrictions on the values, i.e. can they be maps, lists, objects or arrays. If there are restrictions , we should document them. If there are no restrictions we should include examples of more complex objects. 
- in the example is "com.example.User" a "path" in the object to User or just the name of the field.